### DISCUSSION TITLE
how to upgrade laragon without overwriting all my projects

### DISCUSSION THREAD

**Discussion by mbenic** [leokhoa/laragon#877] [Q&A]
hi 
is it possible to install a new version of laragon without overwriting all my projects in the www folder and all the corresponding databases?
thanks for help

**Comment by ppfoong** [comment]
Before installing the new version of Laragon, check for your current version of Apache, PHP and MySQL.

You can do this by viewing the "Version" line of each of them, after right click anywhere in Laragon window and mouse over to the item.

Remember your current version of those components.

Now, install the new version of Laragon to upgrade it.

After that, go to the same place again. When you mouse over to "Version", you should see 2 different versions listed, make sure the old version is selected instead of the new version. Do this for Apache, PHP and MySQL.

That's it.

Note that if you use a new version, old PHP codes might not be compatible and need to be edited. Old database files might also be upgraded for the new version of MySQL, especially when you are upgrading from MySQL 5 to MySQL 8, the old database folder is still there, but a new database folder for MySQL 8 will be created and used instead. Laragon will automatically copy over your data to the new folder.

By using back the old version of Apache, PHP and MySQL, you can keep your existing codes and databases as-is.

But then, why you want to upgrade Laragon in this case?

You can probably use the new version of Apache together with old version of PHP and MySQL, provided there is no compatibility issue between the new Apache and old PHP. If there is, you will hit with error and unable to start the server.

You can always backup the entire Laragon folder in a zip file or into another folder. In case you screw up anything, just delete the new folder and replace it back with your backup folder.
